Sources/CocoaLumberjack/include/CocoaLumberjack/DDContextFilterLogFormatter.h

```diff
@@ -43,6 +99,7 @@ NS_ASSUME_NONNULL_BEGIN
  * For example, logically separate parts of your app each have a different logging context.
  * Also 3rd party frameworks that make use of Lumberjack generally use their own dedicated logging context.
  **/
+__attribute__((deprecated("Use DDContextAllowlistFilterLogFormatter instead")))
 @interface DDContextWhitelistFilterLogFormatter : NSObject <DDLogFormatter>
```

Comment:
I wonder if we can solve this with a `typedef`? Maybe I've been doing Swift for too long now, but in Swift I'd simply create a `typealias OldProtocol = NewProtocol` and deprecate the `typealias`.

Reply:
If that works, we wouldn't need two implementations...

Reply:
It works, of course, consider `typealias TimeInterval = Double` in Swift.
I guess that @sushichop chose second approach via subclassing/cloning.

Second approach is good for distribution, of course.
But we could achieve the same by adding additional files with "non-inclusive" declarations/type aliases.

```objective-c
/// Non-Inclusive file. 
/// We should add notes about its non-inclusiveness at the header.

/// deprecated annotation
typealias OldSomething = NewSomething

/// deprecated annotation
@interface OldSomething
- (void) someOldMethod; //  { [self someNewMethod]; } in implementation.
@end
```

Reply:
Thanks all,
AFAIK, we can use typealias for type in Objective-C(I think typealias should come from C++ feature.). On the other hand, I often see people use typealias only about blocks type in Objective-C. So I prefer subclassing and chose it as @lolgear said. 

Thanks.

Reply:
It's `typedef` in Objective-C and `typealias` in Swift.

We could combine the two: Have a deprecated `typedef` for the old class name (pointing to the new one) and have an extension interface for the old methods (calling the new ones).

A subclass wouldn't be necessary then, and we can put all the "non-inclusive" old stuff into one file (well two, actually, one header one implementation).

Reply:
I'm a bit confused, I might have worded it wrong😅

For example, I think we can use DDContextDenylistFilterLogFormatter itself as a typealias for DContextBlacklistFilterLogFormatter because it is a type(not method and protocol). 
However, DDContextBlacklistFilterLogFormatter has addToBlacklist: method.

I thought I couldn't use DDContextDenylistFilterLogFormatter:addToDenlylist as a typealias for 
DDContextBlacklistFilterLogFormatter:addToBlacklist. We can only call DDContextDenylistFilterLogFormatter:addToDenlylist inside DDContextBlacklistFilterLogFormatter:addToBlacklist.

And we need still provide DDContextBlacklistFilterLogFormatter:addToBlacklist method. It is deprecated method.

Is this wrong?

Reply:
1. typedef BlacklistFormatter = DenylistFormatter
2. All methods with "Blacklist" should have implementation as follows: blackListMethod { denyListMethod }
3. Deprecate typedef 
4. Deprecate all blacklist methods.

Final steps

1. Put everything in CocoaLumberjack+NonInclusive.h(m)
2. Change Podspec/Package.swift if needed.

Reply:
Actually, you could use "method aliasing" via NSProxy, but don't, please. It is easy to start and hard to finish. Something like NSString(cmd.selector).replace("BlackList", "DenyList").toSelector. ( Sorry, hard to write from phone ).

Reply:
> We could combine the two: Have a deprecated `typedef` for the old class name (pointing to the new one) and have an extension interface for the old methods (calling the new ones).

Thanks. And I overlooked this.
I'll do it.

Reply:
@lolgear correctly described it. `typedef` the classes and add the old (deprecated) methods in extensions of those classes... I'd put it in `DDContextFilterLogFormatter+Deprecated.h` or `DDContextFilterLogFormatter+NonInclusive.h`, though. The Package.swift doesn't need to be adjusted (as long as the header file is placed alongside the other public headers).

Reply:
Many thanks!🙂

Reply:
@ffried @sushichop 
Last question.

Will it "required" to distribute "different" packages for "all" and for "non-inclusive"?
I mean "should" we create TWO packages ( two targets for cocoa pods ), one of them will include "non-inclusive" header, and another will not include this header?

Reply:
IMHO this would only create unnecessary overhead and again break existing integrations. Since it's broken, existing users will have to fix it - if they want it or not. If it's deprecated, they're nudged into the right direction, but have no immediate todo.

As mentioned in the original issue, I'm all for such changes, as long as we don't break stuff for existing users - giving them a reason to be pissed about this topic, which would counteract what the movement wants to achieve.